pandas/core/resample.py

```diff
@@ -2115,32 +2120,37 @@ def _adjust_dates_anchored(
     closed: Literal["right", "left"] = "right",
     origin: TimeGrouperOrigin = "start_day",
     offset: Timedelta | None = None,
+    unit: str = "ns",
 ) -> tuple[Timestamp, Timestamp]:
     # First and last offsets should be calculated from the start day to fix an
     # error cause by resampling across multiple days when a one day period is
     # not a multiple of the frequency. See GH 8683
     # To handle frequencies that are not multiple or divisible by a day we let
     # the possibility to define a fixed origin timestamp. See GH 31809
-    first = first.as_unit("ns")
-    last = last.as_unit("ns")
+    first = first.as_unit(unit)
+    last = last.as_unit(unit)
     if offset is not None:
-        offset = offset.as_unit("ns")
+        offset = offset.as_unit(unit)
+
+    freq_value = freq.nanos // (
```

Comment:
would it be simpler to do `Timedelta(freq).as_unit(unit)._value`?

Reply:
(i think freq.nanos is a pattern we need to move away from regardless since there is a risk of overflow)

Reply:
good one, thanks!